modules/ossm-cert-manager-update-istio-csr-revisionbased-only.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * service-mesh-docs-main/install/ossm-cert-manager-assembly.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="updating-istio-csr-revision-based-only_{context}"]
+= Updating istio-csr if using RevisionBased update strategy
+
+If you set `spec.updateStrategy` to `RevisionBased` when creating your `Istio` resource, then `istio-csr` requires you to pass all revisions each time you upgrade your `RevisionBased` control plane. To do this, you need to follow this order each time you upgrade:
+
+. Update `istio-csr` deployment with new revision.
+. Update `Istio.spec.version`.
+
+[id="example-update-revision-based-control-plane_{context}"]
+== Example update for RevisionBased control plane
+
+In this example, the `controlplane` is being updated from `v1.23.0` to `1.23.1.`
+
+. Update `istio-csr` deployment with the new revision:
++
+[source, terminal]
+----
+helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+  --install \
+  --namespace cert-manager \
+  --wait \
+  --reuse-values \
+  --set "app.istio.revisions={default-v1-23-0,default-v1-23-1}"
+----
+
+. Remove the revision from your `istio-csr` deployment:
++
+[source, terminal]
+----
+helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+  --install \
+  --namespace cert-manager \
+  --wait \
+  --reuse-values \
+  --set "app.istio.revisions={default-v1-23-1}"
+----
```

Comment:
Sorry for the confusion. You still need to update your `Istio.spec.version = v1.23.1` in between adding the new revision and removing the old one.

It should be something like:
1. Update `istio-csr` deployment.
2. Update `Istio.spec.version.
3. Remove the old revision from your `istio-csr` deployment.

In the upstream docs for 2. we simply have
> `Then you can update your `Istio.spec.version = v1.23.1`

but here we may need to provide a whole yaml example?
```
kind: Istio
metadata:
  name: default
spec:
  version: v1.23.1
```